src/sys/db/TableCreate.hx

```diff
@@ -86,13 +86,28 @@ class TableCreate {
 			}
 			decls.push(quote(f.name)+" "+getTypeSQL(f.t,dbName)+(f.isNull ? "" : " NOT NULL"));
 		}
-		if( dbName != "SQLite" || !hasID )
+		if( !hasID )
```

Comment:
```suggestion
		if( dbName != "SQLite" || !hasID )
```

Reply:
This previous version of the line 89 was likely the correct one.
 
If we get right of the `dbName != "SQLite"` predicate, then for MySQL the PRIMARY KEY declaration will *only* happen when type has no DID, DUId, DBigId fields. 
Instead we want it all the time for MySQL.

It may solve the unit tests for the PR that failed on mysql.